shiny/Data-Ingest/server_files/ingest_workflow_svr.R

```diff
@@ -31,40 +28,103 @@ updateSelectizeInput(session, "InputParentName",  choices = input_names, server
+
+observeEvent(input$createFormatRecord, {
+  updateSelectizeInput(session, "MimetypeNameCurrent", choices = input$MimetypeName, selected = input$MimetypeName, server = TRUE)
+  
+})
+
+####### Update Text Input for fileName ######
+observe({
+updateTextInput(session, "InputName", value = Shared.data$selected_row)
+})
+
+####### Make Inputs List ##########
 
 observeEvent(input$createInput, {
   ## siteID
-  inputsList$siteName <- input$InputSiteName
-  inputsList$siteID <- sites %>% dplyr::filter(sitename %in% inputsList$siteName) %>% pull(id)
-  
+  if(input$InputSiteName == ""){
+    inputsList$siteName <<- ""
+    inputsList$siteID <<- ""
+  }else{
+    inputsList$siteName <<- input$InputSiteName
+    inputsList$siteID <<- sites %>% dplyr::filter(sitename %in% input$InputSiteName) %>% pull(id)
+  }
+    
   ## ParentID
-  inputsList$parentName <- input$InputParentName
-  inputsList$parentID <- inputs %>% dplyr::filter(name %in% inputsList$parentName) %>% pull(id)
-  
+  if(input$InputParentName == ""){
+    inputsList$parentName <<- ""
+    inputsList$parentID <<- NA
+  }else{
+    inputsList$parentName <<- input$InputParentName
+    inputsList$parentID <<- inputs %>% dplyr::filter(name %in% input$InputParentName) %>% pull(id)
+  }
+
   ## FormatID
-  inputsList$formatName <- input$InputFormatName
-  inputsList$formatID <- formats_sub %>% dplyr::filter(name %in% inputsList$formatName) %>% pull(id)
+  if(input$InputFormatName == ""){
+    inputsList$formatName <<- ""
+    inputsList$formatID <<- ""
+  }else{
+    inputsList$formatName <<- input$InputFormatName
+   # inputsList$formatID <<- formats_sub %>% dplyr::filter(name %in% input$InputFormatName) %>% pull(id) IF Format ID is necessary, I need to redesign this line. 
+  }
+  
+  ## Mimetype (should I find the ID as well?)##
+  inputsList$Mimetype <<- input$MimetypeNameCurrent
   
+
   ## Other Info
-  inputsList$Name <- input$InputName
-  inputsList$StartDate <- input$InputStartDate
-  inputsList$StartTime <- input$StartTimeInput
-  inputsList$EndDate <- input$InputEndDate
-  inputsList$EndTime <- input$EndTimeInput
-  inputsList$Timezone <- input$Timezone
-  inputsList$Notes <- input$InputNotes
+  inputsList$Method <<- input$inputMethod
+  inputsList$Name <<- input$InputName
+  inputsList$Path <<- ifelse(inputsList$Method == "DataONE", file.path(newdir_D1, input$InputName), file.path(local_tempdir, input$InputName))
+  inputsList$StartDate <<- input$InputStartDate
+  inputsList$StartTime <<- input$StartTimeInput
+  inputsList$EndDate <<- input$InputEndDate
+  inputsList$EndTime <<- input$EndTimeInput
+  inputsList$Timezone <<- input$Timezone
+  inputsList$Notes <<- input$InputNotes
+  
+  ## Print List
+  output$summInputs <- renderPrint({inputsList})
+})
+
+
+observeEvent(input$testBety, {
+ output$TestPUT <- renderPrint(inputs$List)
+ inputsList$Path <- "/tmp/RtmpolQsqK/local_tempdir/hf103-01-eddy-2000-01.csv"
```

Comment:
this seems like an odd choice for a default

Reply:
NOOOOOOOO. I missed deleting this while testing. :( Hang on.